workflow/controller/agent.go

```diff
@@ -111,6 +112,48 @@ func (woc *wfOperationCtx) createAgentPod(ctx context.Context) (*apiv1.Pod, erro
 	// Intentionally randomize the name so that plugins cannot determine it.
 	tokenVolumeName := fmt.Sprintf("kube-api-access-%s", rand.String(5))
 
+	agentCtrTemplate := apiv1.Container{
+		Command:         []string{"argoexec"},
+		Image:           woc.controller.executorImage(),
+		ImagePullPolicy: woc.controller.executorImagePullPolicy(),
+		Env:             envVars,
+		SecurityContext: &apiv1.SecurityContext{
+			Capabilities: &apiv1.Capabilities{
+				Drop: []apiv1.Capability{"ALL"},
+			},
+			RunAsNonRoot:             pointer.BoolPtr(true),
+			RunAsUser:                pointer.Int64Ptr(8737),
+			ReadOnlyRootFilesystem:   pointer.BoolPtr(true),
+			AllowPrivilegeEscalation: pointer.BoolPtr(false),
+		},
+		Resources: apiv1.ResourceRequirements{
+			Requests: map[apiv1.ResourceName]resource.Quantity{
+				"cpu":    resource.MustParse("10m"),
+				"memory": resource.MustParse("64M"),
+			},
+			Limits: map[apiv1.ResourceName]resource.Quantity{
+				"cpu":    resource.MustParse(env.LookupEnvStringOr("ARGO_AGENT_CPU_LIMIT", "100m")),
+				"memory": resource.MustParse(env.LookupEnvStringOr("ARGO_AGENT_MEMORY_LIMIT", "256M")),
+			},
```

Comment:
Consider parameterizing these?

Reply:
I'm not sure about this. The agent's memory and CPU usage should be well-defined. It actually (and rarely) should not be different of different users set-up, as we use a watch not informer, and fixed size worked pool.

Configuation would hint at bugs.